lang/c++/api/NodeImpl.hh

```diff
@@ -113,15 +113,15 @@ protected:
         return docAttribute_.get();
     }
 
-    void doAddLeaf(const NodePtr &newLeaf) override {
+    void doAddLeaf(const NodePtr &newLeaf) final {
         leafAttributes_.add(newLeaf);
     }
 
     size_t leaves() const override {
         return leafAttributes_.size();
     }
 
-    const NodePtr &leafAt(int index) const override {
+    const NodePtr &leafAt(size_t index) const override {
```

Comment:
it seems you would want to make this final as well? if you are converting the one above?

Reply:
None of the other virtual methods are called from the constructor and hence leaving them `override` is fine.